File: readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string phrase);

int main(void)
{
    //input from user
    string input = get_string("Text: ");

    //calculate number of letters (no symbols)
    int letters = count_letters(input);

    //count number of words
    int words = 0;

    for (int i = 0, n = strlen(input); i < n; i++)
    {
        if (input[i] == 32)
        {
            words = words + 1;
        }
    }
    words = words + 1;


    //count number of sentences

    int sentences = 0;

    for (int i = 0, n = strlen(input); i < n; i++)
    {
        if (input[i] == '.' || input[i] == '?' || input[i] == '!')
        {
            sentences = sentences + 1;
        }
    }


    //calculate grade of text
    float averagelet = ((float) letters / (float) words) * 100;

    float averagewor = ((float) sentences / (float) words) * 100;

    float index = (0.0588 * averagelet) - (0.296 * averagewor) - 15.8;

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }

    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else
    {
        printf("Grade %i\n", (int) round(index));
    }


}

int count_letters(string phrase)
{
    //total number of characters
    int chars = strlen(phrase);

    //count number of nonletters
    int nonletters = 0;

    for (int i = 0, f = strlen(phrase); i < f; i++)
    {
        if (phrase[i] < 97 && phrase[i] > 90)
        {
            nonletters = nonletters + 1;
        }

        if (phrase[i] < 65 || phrase[i] > 122)
        {
            nonletters = nonletters + 1;
        }

    }
    //calculate total letters only
    int letters = chars - nonletters;

    return letters;

}

File: pset6/readability/readability.py
import cs50
import re


def main():
    # input from user
    input = cs50.get_string("Text: ")

    # calculate number of letters (no symbols)
    letters = count_letters(input)

    # count number of words
    words = len(input.split())

    # count number of sentences
    sentences = len(re.findall('[.?!]', input))

    # calculate grade of text (index)
    averagelet = (letters / words) * 100
    averageword = (sentences / words) * 100

    index = (0.0588 * averagelet) - (0.296 * averageword) - 15.8

    if (index >= 16):
        print("Grade 16+")
        
    elif (index < 1):
        print("Before Grade 1")
        
    else:
        print(f"Grade {int(round(index))}")

    print(letters)
    print(words)
    print(sentences)


# function to count letters
def count_letters(phrase):
    counter = 0
    for letter in phrase:
        if (letter.isalpha()):
            counter = counter + 1

    return counter


if __name__ == "__main__":
    main()